OpenRA.Mods.Common/Activities/Air/FlyAttack.cs

```diff
@@ -147,22 +146,15 @@ public override bool Tick(Actor self)
 
 			var minimumRange = attackAircraft.Info.AttackType == AirAttackType.Strafe ? WDist.Zero : attackAircraft.GetMinimumRangeVersusTarget(target);
 
-			// When strafing we must move forward for a minimum number of ticks after passing the target.
-			if (remainingTicksUntilTurn > 0)
-			{
-				Fly.FlyTick(self, aircraft, aircraft.Facing, aircraft.Info.CruiseAltitude);
-				remainingTicksUntilTurn--;
-			}
-
 			// Move into range of the target.
-			else if (!target.IsInRange(pos, lastVisibleMaximumRange) || target.IsInRange(pos, minimumRange))
+			if (!target.IsInRange(pos, lastVisibleMaximumRange) || target.IsInRange(pos, minimumRange))
 				QueueChild(aircraft.MoveWithinRange(target, minimumRange, lastVisibleMaximumRange, target.CenterPosition, Color.Red));
 
 			// The aircraft must keep moving forward even if it is already in an ideal position.
 			else if (!aircraft.Info.CanHover || attackAircraft.Info.AttackType == AirAttackType.Strafe)
 			{
-				Fly.FlyTick(self, aircraft, aircraft.Facing, aircraft.Info.CruiseAltitude);
-				remainingTicksUntilTurn = attackAircraft.Info.AttackTurnDelay;
```

Comment:
Removing this means that `AttackAircraftInfo.AttackTurnDelay` is no longer used. Is this intentional?

This will have implications for the agility/balancing of aircraft attack runs!

Reply:
It was semi-intentional. Trying to still cram in `AttackTurnDelay` was a big mistake in #16945. The idea here is to always strafe until the maximum weapon range is reached, so that the next strafing pass will start at the optimal weapon range. Of course leaving in a parameter that is not used anywhere is not nice.

I think that thinking in terms of ranges and trajectories here is much nicer than an arbitrary tick count. If it turns out that we still need to tweak this knob, I think we should change `AttackTurnDelay` from a tick count to a `WDist` and have it replace `lastVisibleMaximumRange` in `MoveWithinRange` below when explicitly set.

If it saves us an update rule for now, we can do the same with the current `AttackTurnDelay` and use the speed of the aircraft to translate it into a comparable distance. I think it should be pointed out though that the latest changes to this parameter are likely based on false assumptions about the aircraft behaviour and should be taken with a grain of salt.

Reply:
I really like the idea of specifying a strafe distance instead instead of a tick delay. IMO the hassle of an update rule would be worth it here.